src/pages/Home/User/ParticipatePoll.tsx

```diff
@@ -0,0 +1,192 @@
+import { useNavigate, useParams } from "react-router-dom";
+import useLocalStorage from "../../../utils/useLocalStorage";
+import { useEffect, useState } from "react";
+import {
+  Option,
+  Poll,
+  Question,
+  UserResponse,
+  UserPollResponse,
+} from "../../../interfaces/Poll";
+import generateUniqueId from "../../../utils/useUniqueId";
+import { Button, Card, Radio } from "antd";
+import { PATHS } from "../../../utils/Constants";
+
+type PollCardProps = {
+  question: Question;
+  selectedOptionId: number | null;
+  isPollSubmitted: boolean;
+  onAnswerClick: (que: Question, option: Option) => void;
+};
+
+const PollCard = ({
+  question,
+  selectedOptionId,
+  isPollSubmitted,
+  onAnswerClick,
+}: PollCardProps) => {
+  return (
+    <Card
+      key={question.id}
+      title={question.questionText}
+      className="part-poll-card"
+    >
+      <div className="draft-card-content">
+        <div className="draft-card-icons">
+          <Radio.Group value={selectedOptionId} disabled={isPollSubmitted}>
+            {question.options.map((option) => {
+              return (
+                <Radio
+                  className="option-radio"
+                  key={option.id}
+                  value={option.id}
+                  onClick={() => onAnswerClick(question, option)}
+                >
+                  {option.optionText}
+                </Radio>
+              );
+            })}
+          </Radio.Group>
+        </div>
+      </div>
+    </Card>
+  );
+};
+
+const PollList = () => {
+  const [savedPolls, setSavedPolls] = useLocalStorage("savedPolls", []);
+  const emptyPoll: Poll = {
+    id: 0,
+    label: "",
+    status: "draft",
+    createdBy: "",
+    createdAt: new Date().toISOString(),
+    questions: [],
+  };
+  const [currentPoll, setCurrentPoll] = useState<Poll>(emptyPoll);
+  const [userResponses, setUserResponses] = useLocalStorage(
+    "userResponses",
+    []
+  );
+  const [currentLoggedInUser] = useLocalStorage("currentLoggedInUser", []);
+  const { pollId } = useParams();
+  const navigate = useNavigate();
+  console.log(pollId);
+
+  useEffect(() => {
```

Comment:
1. You can use useMemo here.
2. To get the data from local storage do we really need a hook? because you can easily get that data inside useEffect as well if you have some reusable function.

Reply:
Updated code as per comment